### PR TITLE
[ButtonBar] Fix layout behavior when using custom button title font.

### PR DESCRIPTION
**Pull request by jverkoey**
Button frames need to be updated after the font changes because their frames are not dynamically updated in any layoutSubviews pass.

It may be sensible to update layoutSubviews to do the frame calculations in a follow-up change, but this has the potential to be a behaviorally-breaking change so I've filed a follow-up story: https://www.pivotaltracker.com/story/show/156930078

### CODE REVIEW

File: components/ButtonBar/src/MDCButtonBar.m [hunk 1]
```diff
@@ -418,10 +418,23 @@ - (void)setItems:(NSArray<UIBarButtonItem *> *)items {
 - (void)setButtonsTitleFont:(UIFont *)font forState:(UIControlState)state {
   [_defaultBuilder setTitleFont:font forState:state];
 
-  for (UIView *viewObj in _buttonViews) {
+  for (NSUInteger ix = 0; ix < [_buttonViews count]; ++ix) {
```
Comment (romoore) [inline components/ButtonBar/src/MDCButtonBar.m]: Nit/Consistency: Prefer "i" for the loop variable

File: components/ButtonBar/tests/unit/ButtonBarButtonTitleFontTests.swift [hunk 2]
```diff
@@ -75,6 +75,20 @@ class ButtonBarButtonTitleFontTests: XCTestCase {
     }
   }
 
+  func testCustomFontChangesButtonFrames() {
```
Comment (romoore) [inline components/ButtonBar/tests/unit/ButtonBarButtonTitleFontTests.swift:78]: Please make sure there are tests for both the `item.width > 0` and `MDCCGFloatEqual(item.width, 0)` case.  I'm curious how the branching code above is expected to work.

**Review by romoore — CHANGES_REQUESTED** [review]

File: components/ButtonBar/src/MDCButtonBar.m (hunk 1 above)
Comment (jverkoey) [reply components/ButtonBar/src/MDCButtonBar.m]: Done.

**Review by romoore — APPROVED** [review]
Curious, so setting the width makes it possible the text will be truncated.

File: components/ButtonBar/tests/unit/ButtonBarButtonTitleFontTests.swift (hunk 2 above)
Comment (jverkoey) [reply components/ButtonBar/tests/unit/ButtonBarButtonTitleFontTests.swift:78]: Done.

**Comment by jverkoey** [comment]
> Curious, so setting the width makes it possible the text will be truncated.

Correct.

**Pull request merged**